### PR TITLE
NIAD-998: Only Pathology and Screening message types are published to the inbound MESH queue.

### PR DESCRIPTION
**Pull request by matthewdoreilly**
## Description

Only Pathology and Screening message types are published to the inbound MESH queue.

## Jira Ticket

https://gpitbjss.atlassian.net/browse/NIAD-998

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [x] Acceptance Criteria met
- [x] Commit messages are meaningful
- [x] Manually tested
- [x] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [ ] If your pull request depends on any other, please link them in the description
- [ ] main branch is passing/green on CI
- [ ] Add/update any relevant documentation

### CODE REVIEW

File: src/test/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshServiceTest.java [hunk 1]
```diff
@@ -57,16 +57,17 @@
...
+        meshMessage2.setMeshMessageId(MESSAGE_ID2).setWorkflowId(WorkflowId.PATHOLOGY);
 
+        final long scanIntervalInMilliseconds = 6000L;
         meshService = new MeshService(meshClient,
                 meshInboundQueueService,
                 meshMailBoxScheduler,
                 correlationIdService,
                 scanDelayInSeconds,
-                scanIntervalInMilliseconds,
+            scanIntervalInMilliseconds,
```
Comment (andrewwa-kainos) [inline src/test/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshServiceTest.java:70]: Looks like an accidental change

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshService.java [hunk 2]
```diff
@@ -117,4 +133,7 @@ private void processSingleMessage(String messageId) {
         }
     }
 
+    private boolean isPathologyOrScreeningMessage(WorkflowId workflowId) {
+        return workflowId.equals(WorkflowId.PATHOLOGY) || workflowId.equals(WorkflowId.SCREENING);
+    }
```
Comment (andrewwa-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshService.java:138]: Might suggest making this more general in nomenclature. `isSupportedMessage()`, perhaps. The set of supported workflow IDs can then be an extracted piece of class-level knowledge. If the supported workflows would ever change then we'd only have to change the members of the set, not the name of the method too.
```
private static final Set<WorkflowId> SUPPORTED_WORKFLOWS = Set.of(WorkflowId.PATHOLOGY, WorkflowId.SCREENING);
```
```suggestion
    private boolean isSupportedMessage(WorkflowId workflowId) {
        return SUPPORTED_WORKFLOWS.contains(workflowId);
    }
```
I appreciate that the set of supported workflows is very unlikely to change, so might be unnecessary.

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/MessageType.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+import java.util.Arrays;
+
+@Getter
+@RequiredArgsConstructor
+public enum MessageType {
+    PATHOLOGY_NHSACK("NHS001", "NHS Acknowledgement MIG variant number 1"),
```
Comment (graham-andrew) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/MessageType.java:12]: From what I've seen so far, NHSACKs for Pathology or Screening have the same NHS001 Message type, so we could change this to be more generic:

```suggestion
    NHSACK("NHS001", "NHS Acknowledgement MIG variant number 1"),
```

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshService.java [hunk 4]
```diff
@@ -101,14 +103,28 @@ private void processSingleMessage(String messageId) {
...
+                meshClient.acknowledgeMessage(meshMessage.getMeshMessageId());
+                LOGGER.info("Published MeshMessageId={} for inbound processing", meshMessage.getMeshMessageId());
+            } else {
+                LOGGER.info(
+                    "Downloaded MeshMessageId={} has an unsupported MeshWorkflowId={}."
+                        + "It is not a " + MessageType.PATHOLOGY.getCode()
+                        + " or " + MessageType.PATHOLOGY.getCode() + " message,"
+                        + " therefore it has been left in the inbox.",
+                    meshMessage.getMeshMessageId(), meshMessage.getWorkflowId()
+                );
```
[the remark this answers is not in the document]
Comment (andrewwa-kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshService.java:120]: Just noticed this is `PATHOLOGY` twice - so I guess one of them is meant to be `SCREENING`

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/MessageType.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
```
Comment (saileshp-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/MessageType.java:1]: Could we create a unit test for `MessageType` please? 

Something like `WorkflowIdTest.java` or `PersonIdentificationTypeTest.java`

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshService.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (matthewdoreilly) [reply src/main/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshService.java:120]: I only just realised as well the variables are both the same ("PATHOLOGY" and "PATHOLOGY", not "PATHOLOGY" and "SCREENING") 😅

**Review by andrewwa-kainos — APPROVED** [review]

**Review by graham-andrew — APPROVED** [review]
Looks good to me

**Review by saileshp-kainos — APPROVED** [review]

**Pull request merged**